pkg/database/postgres.go

```diff
@@ -0,0 +1,116 @@
+package database
+
+import (
+	"bytes"
+	"os"
+	"os/exec"
+	"strconv"
+
+	"github.com/pkg/errors"
+)
+
+type Postgres struct {
+	connectionString string
+	logFile          string
+}
+
+func NewPostgres(connectionStr, logFile string) (Postgres, error) {
+	readyCmd := pgIsReadyCmd(connectionStr, 30)
+	errorBuff := captureError(readyCmd)
+
+	err := readyCmd.Run()
+	if err != nil {
+		return Postgres{}, errors.Wrapf(err, "failed to access database, error output: %s", errorBuff.String())
+	}
+
+	return Postgres{connectionString: connectionStr, logFile: logFile}, nil
+}
+
+func (p Postgres) Export(path string) error {
+	dumpCmd := pgDumpCustomCmd(p.connectionString, path)
+	err := p.setOutStream(dumpCmd)
+	if err != nil {
+		return errors.Wrap(err, "failed to set output stream")
+	}
+
+	err = dumpCmd.Run()
+	if err != nil {
+		return errors.Wrap(err, "failed to dump Postgres database")
+	}
+
+	return nil
+}
+
+func (p Postgres) Restore(path string) error {
+	restoreCmd := pgRestoreCmd(p.connectionString, path)
+	err := p.setOutStream(restoreCmd)
+	if err != nil {
+		return errors.Wrap(err, "failed to set output stream")
+	}
+
+	err = restoreCmd.Run()
+	if err != nil {
+		return errors.Wrapf(err, "failed to restore Postgres database")
+	}
+
+	return nil
+}
+
+func (p Postgres) setOutStream(cmd *exec.Cmd) error {
+	if p.logFile == "" {
+		cmd.Stderr = os.Stderr
+		return nil
+	}
+
+	file, err := os.Create(p.logFile)
+	if err != nil {
+		return errors.Wrap(err, "failed to create log file")
+	}
+
+	cmd.Stdout = file
+	cmd.Stderr = file
```

Comment:
I think it might be valuable to have separate files for stderr and stdout but I'm 0/5 on that; it's just a thought.

Reply:
Yea, it is fair point. I think we can stay with one file for now and introduce another if such need arises. There are already many flags for such small app so I think there is value to keep it simple.